planner/plannertest/cmp.go

```diff
@@ -0,0 +1,114 @@
+package plannertest
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/flux/planner"
+	"github.com/influxdata/flux/semantic/semantictest"
+)
+
+// CompareLogicalPlans recursively compares two logical query plans.
+// It returns an error if it encounters a non-matching node.
+// TODO(jlapacik): Need a way to consistently order nodes
+func CompareLogicalPlans(want, got []planner.PlanNode, seen map[planner.PlanNode]bool) error {
```

Comment:
I wonder if defining a custom `Comparer` in the `cmp` package would take care of a lot of the compare logic here?

Reply:
I tried defining a custom `Comparer` however, in the event that two plans aren't the same, the above function will give a much richer diff than a custom `Comparer`.